src/main.ts

```diff
@@ -58,7 +58,7 @@ export default class CopilotPlugin extends Plugin {
                 // @ts-expect-error, not typed
                 const editorView = leaf.view.editor.cm as EditorView;
                 eventListener.onViewUpdate(editorView);
-                eventListener.handleFilePathChange(leaf.view.file.path);
+                eventListener.handleFileChange(leaf.view.file);
             }
         });
```

Comment:
I think you will also need another event. Since if some now removes/adds a tag. The use will have to change from  file before the change is made. 
Can you add the following event listener in `main.ts` to solve this:
```
        this.app.metadataCache.on("changed", (file: TFile) => {
            console.log("File modified", file);
            console.log("instanceof TFile");
            eventListener.handleFileChange(file);
        });
```

Reply:
Done! Should I split `handleFileChange` into `handleFilePathChange` and `handleFileContentChange`, so that the plugin doesn't need to check `isCurrentFilePathIgnored` every time the metadata cache is updated?

It's not an expensive operation by any means, and it looks like a metadata cache change doesn't happen that frequently, but it is perhaps still a little unnecessary. :D

Reply:
As far as I understand, `this.app. metadata cache.on("changed,"...)` gets triggered when you save a file on every keystroke or something like that. I think those are rare enough that it is fine as it is now since it is not an expensive operation for the amount of time it occurs.

Reply:
No worries, sounds good! :-)